spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -224,6 +315,20 @@ data class SpdxLicenseWithExceptionExpression(
         } ?: throw SpdxException("'$exception' is not a valid SPDX license exception id.")
     }
 
+    override fun equals(other: Any?) =
+        when (other) {
+            is SpdxLicenseWithExceptionExpression -> license == other.license && exception == other.exception
+            is SpdxExpression -> {
+                val decomposed = other.decompose()
+                decomposed.size == 1 && decomposed.first().let {
```

Comment:
How about replacing `decomposed.size == 1 && decomposed.first()` with `singleOrNull()`, like

                other.decompose().singleOrNull()?.let {
                    it is SpdxLicenseWithExceptionExpression && it.license == license && it.exception == exception
                } ?: false

Similar in two more places below.

Reply:
I find the underlying condition a bit easier to understand the way it is.